upath/tests/test_core.py

```diff
@@ -137,3 +137,24 @@ def test_create_from_type(path, storage_options, module, object_type):
     except (ImportError, ModuleNotFoundError):
         # fs failed to import
         pass
+
+
+def test_pickling():
+    path = UPath("s3://bucket/folder", storage_options={"anon": True})
```

Comment:
S3 tests are currently skipped in the CI/CD for this so `s3fs` isn't installed. If you change these paths to 'gcs://' they should pass.